lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/Converter.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.line.grok;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.hive.formats.line.grok.exception.GrokException;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Convert String argument to the right type.
+ *
+ * @author anthonyc
+ *
+ */
+// Note: this code is forked from oi.thekraken.grok.api
+// Copyright 2014 Anthony Corbacho, and contributors.
+public class Converter
```

Comment:
While we are cleaning stuff up, can we rename this to `ConverterFactory` and then rename `IConverter` to just `Converter` to avoid the outdated naming convention.

Reply:
Also this class should be final